### PR TITLE
vehicle mechs 2020 (now in 2021, almost 2022)

### PR DESCRIPTION
**Pull request by lolman360**
## About The Pull Request

vehicle mechs

2-person mech support

ripley now has a cage cockpit by default and is not useless for cargo (it is very speedy)

## Why It's Good For The Game

/OBJ/MECHA IS DEAD

long live /obj/vehicle/sealed/mecha

no ability for pilot

## Changelog
:cl: tiviplus, lolman360
add: ripley mkii
refactor: mechs > vehicle/sealed/mecha
/:cl:

### CODE REVIEW

**Comment by shellspeed1** [comment]
Does this mean we are getting the savannah ivanov from tg?

**Comment by lolman360** [comment]
probably not since i dont like it (it's going to be replaced with a different 2-person mech)

**Comment by shellspeed1** [comment]
So long as the option to port it exists I suppose.

File: code/modules/vehicles/mecha/combat/durand.dm [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+	. = ..()
+	if(shield)
+		shield.forceMove(loc)
+		shield.setDir(dir)
+
+/obj/vehicle/sealed/mecha/combat/durand/forceMove(turf/T)
+	. = ..()
+	shield.forceMove(T)
+
+/obj/vehicle/sealed/mecha/combat/durand/mob_exit(mob/M, silent, randomstep, forced)
```
Comment (DeltaFire15) [inline code/modules/vehicles/mecha/combat/durand.dm:53]: This part here causes the linter fail (welln apecifically the line below due to some actions javing sleeps and that proc being should_not_sleep)

File: code/modules/vehicles/mecha/mecha_actions.dm [hunk 2]
```diff
@@ -0,0 +1,275 @@
+//////////////////////MECHA ACTIONS\\\\\\\\\\\\\\\\\\\\\
```
Comment (DeltaFire15) [inline code/modules/vehicles/mecha/mecha_actions.dm:1]: Funky warning

**Comment by rossark** [comment]
the new ripley cannot move diagonally and the system controls menu is able to be used when outside the ripley as well ( for example i could detach modules while outside the mecha)
i couldnt figure out how to make the second ripley thats pressure proof i dont think its possible with plasteel and theres no conversion kit which tg has or maybe im just dumb and missing something
unrelated i think? but i made a gygax and once i got out i still had the green aim reticle hud cursor even outside it

**Comment by dapnee** [comment]
firefighters can't be made now, can't put firefighter suits on the assembled parts plus the sprite disappears.

Mk 1 ripley's don't have an eject button????

**Comment by rossark** [comment]
> Mk 1 ripley's don't have an eject button????

o ye its not written but you cant eject anymore instantly just resist out with a timer

**Comment by shellspeed1** [comment]
Couple of things. First as part of this PR I noticed that the medical gygax was being removed. Is this going to be part of the PR or is it going to be readded? Second, it appears that diagonal movement is now impossible. Are we going to get that back? On another topic, iirc citadel retained the phazon using any anomaly core in its construction, someone correct me if I am wrong. Is this a balance change you are making in this PR? Lastly, is the firefighter being completely removed since we are getting the clarke or will the firefighter remain?

**Comment by lolman360** [comment]
> 
> 
> Couple of things. First as part of this PR I noticed that the medical gygax was being removed. Is this going to be part of the PR or is it going to be readded? Second, it appears that diagonal movement is now impossible. Are we going to get that back? On another topic, iirc citadel retained the phazon using any anomaly core in its construction, someone correct me if I am wrong. Is this a balance change you are making in this PR? Lastly, is the firefighter being completely removed since we are getting the clarke or will the firefighter remain?

medigax removal is unintentional, diagonal movement was unintentional (but imo mechs sholdn't have it anyways), phazon is unintentional too and firefighter stays (we're not getting the clarke)

**Comment by lolman360** [comment]
> Mk 1 ripley's don't have an eject button????

all mechs dont you resist out like vehicles

**Pull request closed without merging**